### ISSUE TITLE
order miss in join query

### ISSUE THREAD

**Issue by f5cs** [ujjwalguptaofficial/JsStore#101]
Order By is miss in join query,
raw code is:
function Join(query, onSuccess, onError) {
      .....
      .....
        if (!_this.errorOccured) {
             _//here miss the order by words._
            var selectObject = new _instance__WEBPACK_IMPORTED_MODULE_1__["Instance"]({
                from: _this.queryStack[0].table,
                where: _this.queryStack[0].where
            },function (results) {...}
}

if possible change to:
function Join(query, onSuccess, onError) {
      .....
      .....
        if (!_this.errorOccured) {
             **var qeury0 = ({
                from: _this.queryStack[0].table,
                where: _this.queryStack[0].where
            }
            if(_this.queryStack[0].order){
               qeury0.order = _this.queryStack[0].order;
           }**
            var selectObject = new _instance__WEBPACK_IMPORTED_MODULE_1__["Instance"](**qeury0**, function (results) {...}
}

**Comment by ujjwalguptaofficial** [comment]
The order wont work there. You can add order in particular select query of join - inside table1 query or table2 query

e.g - 

```
var joinLogic = {
    table1: {
        table: table1_name,
        column: table1.common_field,
        where: {
            Column1: some_value
        },
        order : {
        }
    },
    join: 'inner',
    table2: {
        table: table2_name,
        column: table2.common_field,
        where: {
            Column1: some_another_value
        }
    }
}

```

Hope i m making sense

**Comment by f5cs** [comment]
I did what you said before. But when these order conditions are passed to the backend of the JS store, the jsstore drops the sorting conditions.
u can find this issue in the jsstore.worker.js on row  line 4290 ,u raw code is:
function Join(query, onSuccess, onError) {
.....
.....
if (!this.errorOccured) {
//here miss the order by words.
var selectObject = new instance__WEBPACK_IMPORTED_MODULE_1["Instance"]({
from: _this.queryStack[0].table,
where: _this.queryStack[0].where
},function (results) {...}
}

if possible change to:
function Join(query, onSuccess, onError) {
.....
.....
if (!this.errorOccured) {
var qeury0 = ({
from: _this.queryStack[0].table,
where: _this.queryStack[0].where
}
if(_this.queryStack[0].order){
qeury0.order = _this.queryStack[0].order;
}
var selectObject = new instance__WEBPACK_IMPORTED_MODULE_1["Instance"](qeury0, function (results) {...}
}

**Comment by ujjwalguptaofficial** [comment]
still i m not able to understand the issue exactly. How about you create an example on idbstudio & send me the link ? or you can create a simple project on github . Anyone will work.

Thanks

**Comment by f5cs** [comment]
you can find example width https://github.com/f5cs/jsstore-debug.

**Comment by ujjwalguptaofficial** [comment]
@f5cs - issue is resolved in v - 2.11.1. Hope it works for you.

**Comment by f5cs** [comment]
good job
